File: 03-strings/09_alternating_chars.c
#include "../test_runner.h"

int alternating_characters(char* s) {
    return -1;
}

int main() {
    TEST_INT("AAAA",     alternating_characters("AAAA"),     3);
    TEST_INT("BBBBB",    alternating_characters("BBBBB"),    4);
    TEST_INT("ABABABAB", alternating_characters("ABABABAB"), 0);
    TEST_INT("AAABBB",   alternating_characters("AAABBB"),   4);
    TEST_SUMMARY();
}

File: 03-strings/09_alternating_chars.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Alternating Characters (HackerRank) -- Easy
# Count deletions to make no two adjacent characters the same.
#
#   Key: Count consecutive duplicates

def alternating_characters(s):
    """Return min deletions so no two adjacent are same"""
    pass


run_tests(alternating_characters, [
    (("AAAA",),   3),
    (("BBBBB",),  4),
    (("ABABABAB",),0),
    (("BABAB",),  0),
    (("AAABBB",), 4),
])
